src/main/java/com/scalar/cassy/server/RemoteCommandHandler.java

```diff
@@ -30,7 +31,11 @@ public void run() {
     while (true) {
       RemoteCommandContext future = null;
       try {
-        future = futures.take();
+        future = futures.peek();
```

Comment:
Since the queue size is used to identify if backup is being taken or not, it now does `peek` then `remove`  instead of `take`.